services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableEntityManagerImpl.java

```diff
@@ -1061,4 +1071,41 @@ public Iterator<TableRowChange> newTableRowChangeWithFileRefsIterator(org.sagebi
 		return new PaginationIterator<TableRowChange>((long limit, long offset) -> tableRowTruthDao.getTableRowChangeWithFileRefsPage(idRange, limit, offset), PAGE_SIZE_LIMIT);
 	}
 
+
```

Comment:
Multiple workers could call this at the same time, need to wrap into exclusive lock that still allows read.